ibm_cloud_sdk_core/authenticators/iam_request_based_authenticator.py

```diff
@@ -0,0 +1,113 @@
+# coding: utf-8
+
+# Copyright 2019 IBM All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict
+
+from requests import Request
+
+from .authenticator import Authenticator
+
+class IAMRequestBasedAuthenticator(Authenticator):
+    """The IAMRequestBasedAuthenticator class contains code that is common to all authenticators
+    that need to interact with the IAM tokens service to obtain an access token.
+
+    The bearer token will be sent as an Authorization header in the form:
+
+        Authorization: Bearer <bearer-token>
+
+    Attributes:
+        token_manager (TokenManager): Retrives and manages IAM tokens from the endpoint specified by the url.
+    """
+    authentication_type = 'iam'
```

Comment:
This won't apply to all IAM-based authenticators, so it probably belongs only in the IamAuthenticator class.